hs_tracking/models.py

```diff
@@ -0,0 +1,147 @@
+from json import loads, dumps
+from datetime import datetime, timedelta
+
+from django.db import models
+from django.core import signing
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+from theme.models import UserProfile
+
+
+SESSION_TIMEOUT = 60 * 15
+PROFILE_FIELDS = ["title", "user_type", "subject_areas", "public", "state", "country"]
+USER_FIELDS = ["username", "email", "first_name", "last_name"]
+VISITOR_FIELDS = ["id"] + USER_FIELDS + PROFILE_FIELDS
+if set(PROFILE_FIELDS) & set(USER_FIELDS):
+    raise ImproperlyConfigured("hs_tracking PROFILE_FIELDS and USER_FIELDS must not contain overlapping field names")
+
+
+class SessionManager(models.Manager):
+    def for_request(self, request):
+        signed_id = request.session.get('hs_tracking_id')
+        if signed_id:
+            tracking_id = signing.loads(signed_id)
+            cut_off = datetime.now() - timedelta(seconds=SESSION_TIMEOUT)
+            session = None
+            try:
+                session = Session.objects.filter(begin__gte=cut_off).get(id=tracking_id['id'])
```

Comment:
@ironfroggy I am not sure whether this 15 minutes SESSION_TIMEOUT hard setting takes into account user inactivity for the duration of SESSION_TIMEOUT. If a user has actively browsed/worked on Hydroshare site for more than 15 minutes, then this code will create another session for this user, right? Should we keep the same session for this user instead? Should we check for 'last_activity' to determine whether this session will expire or not?